src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/AbstractPtvProvider.java

```diff
@@ -52,5 +84,565 @@
+      }
+    }
+
+    return result;
+  }
+  
+  protected WebPage translateWebPage(fi.otavanopisto.restfulptv.client.model.WebPage ptvWebPage) {
+    if (ptvWebPage == null) {
+      return null;
+    }
+
+    WebPage webPage = new WebPage();
+    webPage.setLanguage(ptvWebPage.getLanguage());
+    webPage.setType(ptvWebPage.getType());
+    webPage.setUrl(ptvWebPage.getUrl());
+    webPage.setValue(ptvWebPage.getValue());
+    webPage.setDescription(ptvWebPage.getDescription());
+    
+    return webPage;
+  }
+  
+  private List<ServiceHour> translateServiceHours(List<fi.otavanopisto.restfulptv.client.model.ServiceHour> ptvServiceHours) {
+    if (ptvServiceHours == null) {
+      return Collections.emptyList();
+    }
+
+    List<ServiceHour> result = new ArrayList<>(ptvServiceHours.size());
+
+    for (fi.otavanopisto.restfulptv.client.model.ServiceHour ptvServiceHour : ptvServiceHours) {
+      ServiceHour serviceHour = translateServiceHour(ptvServiceHour);
+      if (serviceHour != null) {
+        result.add(serviceHour);
+      }
+    }
+
+    return result;
+  }
+  
+  private ServiceHour translateServiceHour(fi.otavanopisto.restfulptv.client.model.ServiceHour ptvServiceHour) {
+    if (ptvServiceHour == null) {
+      return null;
+    }
+    
+    ServiceHour result = new ServiceHour();
+    result.setAdditionalInformation(translateLanguageItems(ptvServiceHour.getAdditionalInformation()));
+    result.setCloses(ptvServiceHour.getCloses());
+    result.setDays(ptvServiceHour.getDays());
+    result.setOpens(ptvServiceHour.getOpens());
+    result.setStatus(ptvServiceHour.getStatus());
+    result.setType(ptvServiceHour.getType());
+    result.setValidFrom(ptvServiceHour.getValidFrom());
+    result.setValidTo(ptvServiceHour.getValidTo());
+
+    return result;
+  }
+  
+  private List<SupportContact> translateSupportContacts(List<Support> ptvSupportContacts) {
+    if (ptvSupportContacts == null) {
+      return Collections.emptyList();
+    }
+
+    List<SupportContact> result = new ArrayList<>(ptvSupportContacts.size());
+
+    for (Support ptvSupportContact : ptvSupportContacts) {
+      SupportContact supportContact = translateSupportContact(ptvSupportContact);
+      if (supportContact != null) {
+        result.add(supportContact);
+      }
+    }
+
+    return result;
+  }
+  
+  private SupportContact translateSupportContact(Support ptvSupport) {
+    if (ptvSupport == null) {
+      return null;
+    }
+
+    SupportContact support = new SupportContact();
+    support.setEmail(ptvSupport.getEmail());
+    support.setLanguage(ptvSupport.getLanguage());
+    support.setPhone(ptvSupport.getPhone());
+    support.setPhoneChargeDescription(ptvSupport.getPhoneChargeDescription());
+    support.setServiceChargeTypes(ptvSupport.getServiceChargeTypes());
+
+    return support;
+  }
+  
+
+  private List<ServiceChannelAttachment> translateAttachments(List<fi.otavanopisto.restfulptv.client.model.Attachment> ptvAttachments) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateAttachments" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)